creator/creator.go

```diff
@@ -559,15 +571,21 @@ func (c *Creator) Draw(d Drawable) error {
 		return err
 	}
 
-	for idx, blk := range blocks {
+	for idx, block := range blocks {
 		if idx > 0 {
 			c.NewPage()
 		}
 
-		p := c.getActivePage()
-		err := blk.drawToPage(p)
-		if err != nil {
-			return err
+		page := c.getActivePage()
+		if pageBlock, ok := c.pageBlocks[page]; ok {
+			if err := pageBlock.mergeBlocks(block); err != nil {
+				return err
+			}
+			if err := mergeResources(block.resources, pageBlock.resources); err != nil {
```

Comment:
Should this be done inside the mergeBlocks function?

Reply:
It could be done there. However, mergeBlocks is called from some of the components and we don't need to call the function there. It would decrease performance a bit.